### PR TITLE
Period.Between for YearMonth

### PR DESCRIPTION
**Pull request by lilinus**

### CODE REVIEW

**Comment by jskeet** [comment]
Thanks for this - will review at the weekend.

File: src/NodaTime/Period.cs [hunk 1]
```diff
@@ -715,6 +715,59 @@ internal static int InternalDaysBetween(LocalDate start, LocalDate end)
...
+        /// If <paramref name="end"/> is before <paramref name="start" />, each property in the returned period
+        /// will be negative. If the given set of units cannot exactly reach the end point (e.g. finding
+        /// the difference between February 2010 and March 2012 in years) the result will be such that adding it to <paramref name="start"/>
+        /// will give a value between <paramref name="start"/> and <paramref name="end"/>. In other words,
+        /// any rounding is "towards start"; this is true whether the resulting period is negative or positive.
+        /// </remarks>
+        /// <param name="start">Start year and month</param>
+        /// <param name="end">End year and month</param>
+        /// <param name="units">Units to use for calculations</param>
+        /// <exception cref="ArgumentException"><paramref name="units"/> contains time units, is empty or contains unknown values.</exception>
```
Comment (malcolmr) [inline src/NodaTime/Period.cs:731]: It's not just "time units" here that are disallowed: `Days` is also prohibited.

File: src/NodaTime/Period.cs [hunk 2]
```diff
@@ -715,6 +715,59 @@ internal static int InternalDaysBetween(LocalDate start, LocalDate end)
             return endDays - startDays;
         }
 
+        /// <summary>
+        /// Returns the period between a start and an end YearMonth, using only the given units.
```
Comment (malcolmr) [inline src/NodaTime/Period.cs:719]: This probably should be `<see cref="YearMonth"/>`.

File: src/NodaTime/Period.cs [hunk 3]
```diff
@@ -715,6 +715,59 @@ internal static int InternalDaysBetween(LocalDate start, LocalDate end)
...
+                return Zero;
+            }
+
+            // Multiple fields
+            DateComponentsBetween(start.StartDate, end.StartDate, units, out int years, out int months, out _, out _);
+            return new Period(years, months, 0, 0);
+        }
+
+        /// <summary>
+        /// Returns the exact difference between two YearMonths.
```
Comment (malcolmr) [inline src/NodaTime/Period.cs:755]: (perhaps similarly here?)

**Review by malcolmr — COMMENTED** [review]
LGTM, fwiw.

**Review by jskeet — APPROVED** [review]
Looks great, thanks!

**Pull request merged**